nav2_collision_monitor/src/polygon.cpp

```diff
@@ -450,6 +438,25 @@ bool Polygon::getParameters(
   return true;
 }
 
+void Polygon::createSubscription(std::string & polygon_sub_topic)
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  if (!polygon_sub_topic.empty()) {
+    RCLCPP_INFO(
+      logger_,
+      "[%s]: Subscribing on %s topic for polygon",
+      polygon_name_.c_str(), polygon_sub_topic.c_str());
```

Comment:
Can't you use the same `polygon_sub_topic` and when this is a polygon do this operation with the `PolygonStamped` type and when using circle use the `Float32` type? That would remove the need for some or all of the changes in  `Circle::getParameters`

Reply:
We can use inheritance to our benefit here :-)

Reply:
I agree it's a better design to keep same parameter `polygon_sub_topic` for both polygon and circle types! 

I moved declaration and init of parameters `polygon_sub_topic` and `footprint_topic` to function `Polygon::getCommonParameters()`. So that in function `Circle::getParameters` there is only the declaration and init of parameter `radius`.